build_defs/java.build_defs

```diff
@@ -87,6 +90,33 @@ def java_library(name:str, srcs:list=None, src_dir:str=None, resources:list=[],
 
         src_dir_label = [] if srcs else ['src_dir:'+src_dir]
 
+        def _jni_header_outputs(name, output):
+            for hdr_path in output:
+                if hdr_path != "":
+                    add_out(name, hdr_path)
+                    add_label(name, "cc:inc:" + join_path(package_name(), dirname(hdr_path)))
+
+        jni_hdrs = build_rule(
+            name = name,
+            tag = "jni_hdrs",
+            srcs = {
+                "WORKER": srcs or [src_dir],
+            },
+            cmd = " && ".join([
+                "mkdir -p _jni",
+                f"{javac_cmd} -h _jni >/dev/null 2>&1",
```

Comment:
Silencing the output of `javac` here as a precaution because it's possible to instruct `javac` to output to stdout instead of stderr using a command line option (i.e. via `javac_flags`) - it's an edge case, but one that'll break this target if it's used. The main target will still show compilation errors for the class files.